crates/configuration/src/configuration.rs

```diff
@@ -282,14 +282,10 @@ fn get_primary_key_uniqueness_constraint(
     name: &str,
     collection_type: &str,
 ) -> Option<(String, ndc::UniquenessConstraint)> {
-    // Check to make sure our collection's object type contains the _id objectid field
+    // Check to make sure our collection's object type contains the _id field
     // If it doesn't (should never happen, all collections need an _id column), don't generate the constraint
     let object_type = object_types.get(collection_type)?;
-    let id_field = object_type.fields.get("_id")?;
-    match &id_field.r#type {
-        schema::Type::Scalar(BsonScalarType::ObjectId) => Some(()),
-        _ => None,
-    }?;
+    let _id_field = object_type.fields.get("_id")?;
```

Comment:
Shouldn't this be validated to at least be a scalar type (and shouldn't it only be scalar types with equality?)? 
What's the behaviour if the _id column is an object type? Does that break in v3-engine?

Reply:
MongoDB itself doesn't care. The `_id` can have any type and all types are comparable. However, I don't want to potentially break the engine, so I've modified it to only allow scalar types for which we define an equality operator in the capabilities.